packages/cli/src/utils/get-project-info.ts

```diff
@@ -41,22 +38,68 @@ const TS_CONFIG_SCHEMA = z.object({
   }),
 })
 
+export async function detectFrameworkConfigFiles(cwd: string): Promise<Framework | null> {
+  const packageInfo = await getPackageInfo(cwd, false)
+  const configFiles = await glob('**/{nuxt,vite,astro}.config.*|composer.json', {
+    cwd,
+    deep: 3,
+    ignore: PROJECT_SHARED_IGNORE,
+  })
+
+  // Check for Nuxt
+  if (configFiles.find(file => file.startsWith('nuxt.config.'))) {
+    const isUsingAppDir = await fs.pathExists(path.resolve(cwd, 'app'))
+    return isUsingAppDir ? FRAMEWORKS.nuxt4 : FRAMEWORKS.nuxt3
+  }
```

Comment:
@sadeghbarati There is a problem with this detection approach. Having the `app` folder does not necessarily mean it is nuxt4, since nuxt3 (with `compatibilityVersion: 4`) also uses the `app` folder structure.

Instead, we should probably check the `package.json` for the exact nuxt version using semver, which is what nuxi is using.

https://github.com/nuxt/cli/blob/77b68177d574e302c91dfc86137e9b9aaa95343f/packages/nuxi/src/utils/versions.ts#L1-L12

Reply:
Thanks ❤️